packages/cli/src/services/linter/utils/getRuleset.ts

```diff
@@ -2,8 +2,12 @@ import { Optional } from '@stoplight/types';
 import { Ruleset, RulesetDefinition } from '@stoplight/spectral-core';
 import * as fs from 'fs';
 import * as path from '@stoplight/path';
-import { isAbsolute } from '@stoplight/path';
 import * as process from 'process';
+import { extname } from '@stoplight/path';
+import { migrateRuleset } from '@stoplight/spectral-ruleset-migrator';
+
+// eslint-disable-next-line @typescript-eslint/require-await
+const AsyncFunction = (async (): Promise<void> => void 0).constructor as FunctionConstructor;
```

Comment:
Do I even want to know?

Reply:
So, long story short - in rare circumstances a "top-level" await might be inserted.
This won't be a problem soon-ish once we hop on ESM, but for the time being it's a plain CommonJS.
The above is needed for us to be able to do

```
AsyncFunction('module.exports = await /* the rest of the code goes here */)'
```

Reply:
Narrator: it turns out he really did not want to know :trollface: